<launch>
  <!-- carla town04 default-->
  <!-- <arg name="init_x" value="258.4" />
  <arg name="init_y" value="206.5" />
  <arg name="init_z" value="0.0" />
  <arg name="init_roll" value="0.0" />
  <arg name="init_pitch" value="0.0" />
  <arg name="init_yaw" value="1.54" /> -->

  <!-- carla town04 circle-->
  <!-- <arg name="init_x" value="314.8" />
  <arg name="init_y" value="230.0" />
  <arg name="init_z" value="0.0" />
  <arg name="init_roll" value="0.0" />
  <arg name="init_pitch" value="0.0" />
  <arg name="init_yaw" value="1.54" /> -->
  <arg name="init_x" value="257.6" />
  <arg name="init_y" value="310.8" />
  <arg name="init_z" value="0.3" />
  <arg name="init_roll" value="0.0" />
  <arg name="init_pitch" value="0.0" />
  <arg name="init_yaw" value="0.0" />

  <!-- vel/pose connect -->
  <arg name="topic_pose_stamped" default="/ndt_pose" />
  <arg name="topic_twist_stamped" default="/estimate_twist" /> 
  <!-- <arg name="topic_twist_stamped" default="/odom_twist" /> -->

  <!-- points downsampler -->
  <include file="$(find points_downsampler)/launch/voxel_grid_filter_params.launch">
    <arg name="leaf_size" value="0.5" />
    <arg name="measurement_range" value="100" />
  </include>

  <!-- <arg name="leaf_size" default="1.0" />
  <arg name="measurement_range" default="200" />

  <node pkg="points_downsampler" name="voxel_grid_filter" type="voxel_grid_filter">
    <param name="input_topic_name" value="points_raw" />
    <param name="output_topic_name" value="filtered_points" />
    <param name="leaf_size" value="$(arg leaf_size)" />
    <param name="measurement_range" value="$(arg measurement_range)" />
  </node> -->


  <!-- ndt matching kalman filter parameters -->
  <rosparam command="load" file="$(env USER_HOME)/rubis_ws/src/rubis_autorunner/cfg/cubetown_autorunner/kalman_filter.yaml" />

  <!-- ndt_matching --><!-- pcl_generic=0, pcl_anh=1, pcl_anh_gpu=2, pcl_openmp=3 -->
  <include file="$(find lidar_localizer)/launch/ndt_matching_params.launch">
    <arg name="method_type" value="1" /> 
    <arg name="get_height" value="true" /> 
  </include>

  <!-- ndt config -->
  <node pkg="rostopic" type="rostopic" name="config_ndt"
  args="pub /config/ndt autoware_config_msgs/ConfigNDT
  '{header: {seq: 8, stamp: {secs: 0, nsecs: 0}, frame_id: ''},
    init_pos_gnss: 0,
    x: $(arg init_x),
    y: $(arg init_y),
    z: $(arg init_z),
    roll: $(arg init_roll), 
    pitch: $(arg init_pitch),
    yaw: $(arg init_yaw),
    use_predict_pose: 1,
    error_threshold: 0.05,
    resolution: 1.0,
    step_size: 0.5,
    trans_epsilon: 0.01,
    max_iterations: 2}
  '"/>

</launch>
